### PR TITLE
Fix copy link issue in Safari

### PR DESCRIPTION
**Pull request by leanneeliatra**
### Description
Bug fix for broken copy link in Safari

### Issues Resolved
- Closes https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1716

### Testing
- Companion PR https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/940

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by peternied** [comment]
@leanneeliatra Please fix the ci failures and see about adding some kind of tests to validate this change.

**Comment by leanneeliatra** [comment]
> @leanneeliatra Please fix the ci failures and see about adding some kind of tests to validate this change.

Unit tests in progress to conclude this ticket (integration test and bugfix complete).

**Comment by peternied** [comment]
@leanneeliatra It looks like there are unintended changes included in this PR, maybe you'd be better of rebasing from main and force pushing again?

**Comment by peternied** [comment]
@leanneeliatra Looks like the rebase issues are behind now, happy to review once the test case is built out - thanks!

**Comment by leanneeliatra** [comment]
> @leanneeliatra Looks like the rebase issues are behind now, happy to review once the test case is built out - thanks!

Thank you Peter the cypress test is now complete for functionality, there is an linting error I am fixing. The jest test in progress. I will update once complete for review.

**Comment by leanneeliatra** [comment]
Linting errors for [corresponding integration tests are now solved, I have some comments to address there ](https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/940) and unit tests in progress.

**Review by scrawfor99 — DISMISSED** [review]
Looks good to me

**Comment by leanneeliatra** [comment]
@peternied could you review this when you have a moment please. Thanks.

**Comment by leanneeliatra** [comment]
The [integration tests](https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/940#pullrequestreview-1749481261) are now reviewed and are waiting on the final review and merge of this PR, to be merged themselves.

cc @peternied @cwperks @DarshitChanpura @peternied @RyanL1997 this PR just needs 1 more review if anyone is available.

**Review by derek-ho — COMMENTED** [review]
No major concerns with this PR, but not able to replicate the original issue, so not sure if this is needed

**Comment by leanneeliatra** [comment]
My Safari Version is 17.1 (19616.2.9.11.7). I checked again and I can indeed reproduce this and my fix does fix it locally. From speaking with @davidlago, in the case that it can't be reproduced by some but I myself can reproduce it as could the original reporter, we can go ahead and merge this if possible please? Thanks all.

**Review by peternied — APPROVED** [review]

File: public/services/shared-link.ts
Comment (cwperks) [inline public/services/shared-link.ts:71]: What is the difference between urlPart and originalValue? From the `processCopyEvent` function it looks like urlPart is the originalValue unless originalValue is an `iframe`. If `originalValue` is an `iframe` then processCopyEvent will do some addition work to extract the `src`. Since the function to `updateClipboard` can sometimes return `originalValue` is there ever a case where it can return the `iframe` instead of the extracted `src`?

Comment (cwperks) [reply public/services/shared-link.ts:71]: Ignore this comment. I just checked the code on `main` and see how its related to the `addTenantToURL` method.

**Review by cwperks — APPROVED** [review]
LGMT! Thank you @leanneeliatra 🚢

**Pull request merged**
